core/src/main/java/hudson/model/Queue.java

```diff
@@ -1460,54 +1460,64 @@ public void maintain() {
             // allocate buildable jobs to executors
             for (BuildableItem p : new ArrayList<BuildableItem>(
                     buildables)) {// copy as we'll mutate the list in the loop
-                // one last check to make sure this build is not blocked.
-                if (isBuildBlocked(p)) {
-                    p.leave(this);
-                    new BlockedItem(p).enter(this);
-                    LOGGER.log(Level.FINE, "Catching that {0} is blocked in the last minute", p);
-                    // JENKINS-28926 we have moved an unblocked task into the blocked state, update snapshot
-                    // so that other buildables which might have been blocked by this can see the state change
-                    updateSnapshot();
-                    continue;
+                if (p.task instanceof FlyweightTask) {
+                    Runnable r = makeFlyWeightTaskBuildable(new BuildableItem(p));
+                    if (r != null) {
+                        p.leave(this);
+                        r.run();
+                        updateSnapshot();
+                    }
                 }
+                else {
+                    // one last check to make sure this build is not blocked.
+                    if (isBuildBlocked(p)) {
+                        p.leave(this);
+                        new BlockedItem(p).enter(this);
+                        LOGGER.log(Level.FINE, "Catching that {0} is blocked in the last minute", p);
+                        // JENKINS-28926 we have moved an unblocked task into the blocked state, update snapshot
+                        // so that other buildables which might have been blocked by this can see the state change
+                        updateSnapshot();
+                        continue;
+                    }
 
-                List<JobOffer> candidates = new ArrayList<JobOffer>(parked.size());
-                for (JobOffer j : parked.values())
-                    if (j.canTake(p))
-                        candidates.add(j);
-
-                MappingWorksheet ws = new MappingWorksheet(p, candidates);
-                Mapping m = loadBalancer.map(p.task, ws);
-                if (m == null) {
-                    // if we couldn't find the executor that fits,
-                    // just leave it in the buildables list and
-                    // check if we can execute other projects
-                    LOGGER.log(Level.FINER, "Failed to map {0} to executors. candidates={1} parked={2}",
-                            new Object[]{p, candidates, parked.values()});
-                    continue;
-                }
+                    List<JobOffer> candidates = new ArrayList<JobOffer>(parked.size());
```

Comment:
Normally it wouldn't show the spaces in a different colour unless they are tabs... if not TABs then that's fine, just check